cv-data-controller/src/test/java/com/trihydro/cvdatacontroller/controller/CdotUpstreamPathControllerTest.java

```diff
@@ -0,0 +1,296 @@
+package com.trihydro.cvdatacontroller.controller;
+
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.trihydro.cvdatacontroller.controller.CdotUpstreamPathController.DistanceCalculator;
+import com.trihydro.cvdatacontroller.controller.CdotUpstreamPathController.PathDirection;
+import com.trihydro.library.helpers.CdotGisConnector;
+import com.trihydro.library.helpers.Utility;
+import com.trihydro.library.model.Milepost;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+
+class CdotUpstreamPathControllerTest extends TestBase<CdotUpstreamPathController> {
```

Comment:
suggestion: to allow use of the constructor instead of InjectDependencies
```suggestion
class CdotUpstreamPathControllerTest {
```

Reply:
Addressed